contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -509,12 +509,38 @@ namespace eosiosystem {
       uint64_t  primary_key() const { return giftee.value; } // unique as one giftee can only hold gifted ram from one gifter
    };
 
+   struct [[eosio::table("peerkeys"), eosio::contract("eosio.system")]] peer_key
+   {
+      struct v0_data {
+         std::optional<eosio::public_key> pubkey;  // peer key for network message authentication
+         EOSLIB_SERIALIZE( v0_data, (pubkey) )
+      };
+
+      name                                           proposer_finalizer_name;
+      uint32_t                                       block_num; // block number where this row was emplaced or modified
+      uint8_t                                        version;   // version == 0 means variant holds `v0_data`
```

Comment:
There is no need for the `uint8_t version` since you already have adopted a `std::variant` for the `data` below.

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/185/commits/551a735627735c6c4e50398bb7d475d858a3c650 & https://github.com/AntelopeIO/spring/pull/1276/commits/ee4eb99e7d145c56508230d0a8df5b6d2ab9370d